### ISSUE TITLE
GeoClue2 only works after waiting 60 seconds

### ISSUE THREAD

**Issue by lunarlattice0** [maltegrosse/go-geoclue2#1]
I am aware that this library hasn't been updated for quite some time, but I am having some trouble with it. If try to get the location (as seen in the example), wait 60 seconds, then try again, GeoClue2 functions as normal.

However, if I attempt to get the location faster than that, the library errors with "timed out gathering location object".

**Comment by maltegrosse** [comment]
@Thelolguy1  I havent played around with geoclue2 since then - but is geoclue running before? also manager and client should somehow run forever, and only if u need the location in your application, you should ask the running client....

**Comment by lunarlattice0** [comment]
Yes, geoclue is running. If I try to get the location in between the 60 seconds, a new client appears momentarily before disappearing. In my situation, the user may close the program in between the 60 seconds meaning location cannot be acquired in that gap.

**Comment by maltegrosse** [comment]
does the initial request takes 60 seconds? once the 60 seconds passed, can you request the location with less time passed by? please share some code snippets

**Comment by lunarlattice0** [comment]
Sure, the code in question is [here](https://github.com/Thelolguy1/phome/blob/master/main.go#L92C3-L92C3). The initial request and the request after 60 seconds is instantaneous.

**Comment by maltegrosse** [comment]
seems like u just run the example code twice?

**Comment by lunarlattice0** [comment]
Yeah, but it doesn't work if I run the example code again within 60 seconds.
